### ISSUE TITLE
09-k-means-housing.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/Gupir11-machine-learning#6]
Hola @Gupir11 , por aquí las issues del ejercicio de clustering!

## LO QUE HAS HECHO BIEN

- **Selección de variables correcta**: Latitude, Longitude y MedInc
- **Uso de StandardScaler**: Alternativa válida a MinMaxScaler
- **Train/test split**: Correcto con test_size=0.2 y random_state
- **K-Means con k=6 y n_init=10**: Parámetros correctos
- **Visualización geográfica**: Scatter plot de Longitude vs Latitude con colores por cluster
- **Comparación train vs test visual**: Buena idea usar alpha diferente y marker="^" para test
- **Random Forest supervisado**: Correcto, usas los labels de K-Means como target
- **Métricas**: accuracy_score y classification_report
- **Observación sobre la predicción**: Correcta interpretación de que los puntos test caen en zonas esperadas

---

### 1. Error: Escalas TODO el dataset ANTES del split (aunque no lo uses)

Luego vuelves a escalar después del split.

El primer `fit_transform(X)` no se usa para nada después. Puedes eliminar esas líneas:

<img width="906" height="574" alt="Image" src="https://github.com/user-attachments/assets/3e143bc9-2030-4b7f-b72d-fc166c52fe26" />

El escalado correcto ya lo haces después del split, así que no es un error grave, solo código redundante que puede confundir.

---

### 2. Falta matriz de confusión

```python
from sklearn.metrics import confusion_matrix
import seaborn as sns

cm = confusion_matrix(y_test, y_pred)
plt.figure(figsize=(8, 6))
sns.heatmap(cm, annot=True, fmt='d', cmap='Blues',
            xticklabels=[f'Cluster {i}' for i in range(6)],
            yticklabels=[f'Cluster {i}' for i in range(6)])
plt.xlabel('Predicción')
plt.ylabel('Real')
plt.title('Matriz de Confusión - Random Forest')
plt.show()
```

---

### 3. Opcional pero importante. Silhouette Score

Métrica importante para evaluar la calidad del clustering:

```python
from sklearn.metrics import silhouette_score

silhouette = silhouette_score(X_train_scaled, kmeans.labels_)
f"Silhouette Score: {silhouette:.4f}"
```

---

### 4. Falta función predict

```python
def predict_cluster(latitude: float, longitude: float, medinc: float) -> int:
    """
    Predice el cluster de una vivienda.
    """
    new_data = [[latitude, longitude, medinc]]
    new_data_scaled = scaler.transform(new_data)
    return kmeans.predict(new_data_scaled)[0]

# Ejemplo
print(f"Cluster: {predict_cluster(37.77, -122.42, 8.5)}")
```

---

### 5. LEVE: Falta guardar los modelos

---

### 6. LEVE: Intenta siempre usar rutas relativas 

<img width="915" height="274" alt="Image" src="https://github.com/user-attachments/assets/c67664dc-531f-4a2f-a8e9-2cd93b7e2434" />
